### PR TITLE
Supporting Documents

### PR DESCRIPTION
**Pull request by EvanBarbour3**
### Added
- Added support to upload a supporting document from a retailer (#158623202)

[Delivers [#158623202](https://www.pivotaltracker.com/story/show/158623202)]

### CODE REVIEW

**Comment by wnowicki** [comment]
@EvanBarbour3 Conflicts!

File: tests/unit/Controllers/SupportingDocumentsControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        $this->actingAs($user, 'api');
+    }
+
+    /**
+     * @author EB
+     */
+    public function testGetDictionary()
+    {
+        $installation = new Installation();
+        $merchant = new Merchant();
```
[the remark this answers is not in the document]
Comment (wnowicki) [reply tests/unit/Controllers/SupportingDocumentsControllerTest.php]: - good question, but is not the rule that when ever we can we use real objects?

File: resources/assets/js/components/applications/documents/Upload.vue [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                    accept: ['image/*','.pdf','.doc','.docx','.docm'],
+                    onUploadDone: (response) => this.documentUploaded(response.filesUploaded),
+                };
+                client.picker(options).open();
+            },
+            documentUploaded (file) {
+                this.getCategory().then(category => {
+                    this.cursorLoading();
+                    axios.post(
+                        '/api/installations/1/applications/20/documents',
```
[the remark this answers is not in the document]
Comment (wnowicki) [reply resources/assets/js/components/applications/documents/Upload.vue]: -is that a bug ? 🐛

[the remark this answers is not in the document]
Comment (KrekoG) [reply resources/assets/js/components/applications/documents/Upload.vue]: yes :)

File: tests/unit/Controllers/SupportingDocumentsControllerTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (KrekoG) [reply tests/unit/Controllers/SupportingDocumentsControllerTest.php]: Ideally we would use the mock of the interface, so the test is implementation agnostic.

The reason I say we should use the mock, is that the test should not care about how the `Merchant` and the `Installation` objects are created, but if we use the real objects, it has to.

[the remark this answers is not in the document]
Comment (KrekoG) [reply tests/unit/Controllers/SupportingDocumentsControllerTest.php]: So to answer your question, no, the rule really is quite the opposite.

[the remark this answers is not in the document]
Comment (EvanBarbour3) [reply tests/unit/Controllers/SupportingDocumentsControllerTest.php]: We can use a mock object but then we'd have to mock it using properties of the object and I'm not sure we can?

**Comment by EvanBarbour3** [comment]
@KrekoG

**Comment by EvanBarbour3** [comment]
@wnowicki

File: tests/unit/Controllers/SupportingDocumentsControllerTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (KrekoG) [reply tests/unit/Controllers/SupportingDocumentsControllerTest.php]: Yeah, don't worry about it, there is no point getting stuck on this.

**Pull request merged**